### PR TITLE
ON-45150 # add CalendarBookingsForm and related components

### PR DESCRIPTION
**Pull request by divporter**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

### CODE REVIEW

File: src/components/calendar-bookings/CalendarBookingsReschedulingForm.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  return (
+    <div className="ob-scheduling-booking-form">
+      <NylasScheduling
+        sessionId={sessionId}
+        eventOverrides={{
+          timeslotConfirmed: onTimeSlotConfirmed,
+          bookedEventInfo: async (event) => {
+            event.preventDefault()
+            console.log(event)
```
Comment (mymattcarroll) [inline src/components/calendar-bookings/CalendarBookingsReschedulingForm.tsx:40]:
```suggestion
```

Comment (divporter) [reply src/components/calendar-bookings/CalendarBookingsReschedulingForm.tsx:40]: Argh, from the other story I was doing

File: src/components/calendar-bookings/CalendarBookingsForm.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        />
+      </div>
+    </>
+  )
+}
+
+function CalendarBookingsForm({
+  postSubmissionAction,
+}: {
+  postSubmissionAction: (
```
Comment (mymattcarroll) [inline src/components/calendar-bookings/CalendarBookingsForm.tsx:191]:
```suggestion
  onDone: (
```

To align with payment components

File: src/components/calendar-bookings/CalendarBookingsForm.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+              timeslotConfirmed: onTimeSlotConfirmed,
+              bookedEventInfo: async (event) => {
+                event.preventDefault()
+
+                if (event.detail.error) {
+                  setBookingError(
+                    event.detail.error.message ?? 'Calendar Booking Error',
+                  )
+                } else {
+                  event.detail
```
Comment (mymattcarroll) [inline src/components/calendar-bookings/CalendarBookingsForm.tsx:129]:
```suggestion
```

File: src/components/calendar-bookings/CalendarBookingsForm.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+      postSubmissionError: null,
+    }))
+  }, [])
+  const clearConfirmingBookingError = React.useCallback(() => {
+    setPostSubmissionState((currentState) => ({
+      ...currentState,
+      confirmingBookingError: null,
+    }))
+  }, [])
+
```
Comment (mymattcarroll) [inline src/components/calendar-bookings/CalendarBookingsForm.tsx:68]: This file should also include the `executePostSubmissionAction()` that is in the renderer as we want the error handling and such. We should be called `onDone()` within that function instead of `submissionService.executePostSubmissionAction()`.

File: src/components/calendar-bookings/CalendarBookingsForm.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+          >
+            <div className="buttons">
+              <button
+                type="button"
+                className={clsx(
+                  'is-primary button ob-button ob-scheduling-receipt__button ob-scheduling-receipt__okay-button cypress-scheduling-receipt-okay-button',
+                  { 'is-loading': isRunningPostSubmissionAction },
+                )}
+                disabled={isRunningPostSubmissionAction}
+                onClick={() => onDone(formSubmissionResult)}
```
Comment (mymattcarroll) [inline src/components/calendar-bookings/CalendarBookingsForm.tsx:192]:
```suggestion
                onClick={() => executePostSubmissionAction(formSubmissionResult)}
```
